stream-chat-android-client/src/main/java/io/getstream/chat/android/client/notifications/LoadNotificationDataWorker.kt

```diff
@@ -0,0 +1,143 @@
+package io.getstream.chat.android.client.notifications
+
+import android.app.Notification
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.content.Context
+import android.os.Build
+import androidx.core.app.NotificationCompat
+import androidx.work.CoroutineWorker
+import androidx.work.ExistingWorkPolicy
+import androidx.work.ForegroundInfo
+import androidx.work.OneTimeWorkRequestBuilder
+import androidx.work.WorkManager
+import androidx.work.WorkerParameters
+import androidx.work.workDataOf
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.client.R
+import io.getstream.chat.android.client.logger.ChatLogger
+
+internal class LoadNotificationDataWorker(
+    private val context: Context,
+    workerParams: WorkerParameters,
+) : CoroutineWorker(context, workerParams) {
```

Comment:
What's the delay here (especially in doze mode)?

Reply:
works almost immediately on my phone (S10, Android 11)